Assets/Scripts/receiveSignal/receiveSignalGhostWall.cs

```diff
@@ -0,0 +1,42 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class receiveSignalGhostWall : receiveSignal {
+	private GhostWall[] ghostWalls;
+
+	void Start(){
+		ghostWalls = GetComponentsInChildren<GhostWall>();
+	}
+
+	public override void activate(){
+		foreach (GhostWall wall in ghostWalls) {
+			activateChild (wall.GetComponent<Collider2D> (), wall.GetComponent<Renderer> ());
+		}
+	}
+
+	public override void deactivate(){
+		foreach (GhostWall wall in ghostWalls) {
+			deactivateChild (wall.GetComponent<Collider2D> (), wall.GetComponent<Renderer> ());
+		}
+	}
+
+	public void activateChild(Collider2D collider2D2, Renderer renderer2){
```

Comment:
Shouldn't these methods maybe be in a script for the child walls? Because they are more specific to the child elements.